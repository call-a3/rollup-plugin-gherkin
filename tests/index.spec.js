import { rollup } from 'rollup'
import hypothetical from 'rollup-plugin-hypothetical'

import rollupPluginGherkin from '../src'

test('should compile Gherkin code correctly', async () => {
  const bundle = await rollup({
    input: './test/gherkin.feature',
    plugins: [
      hypothetical({
        files: {
          './test/gherkin.feature': `Feature: Background

  Background: a simple background
    Given the minimalism inside a background


  Scenario: minimalistic
    Given the minimalism
    When a minimalist is minimalistic
    Then minimalism is minimally achieved
`,
        },
      }),
      rollupPluginGherkin(),
    ],
  })

  const { code, map } = await bundle.generate({ format: 'es', sourcemap: true })

  expect(code).toMatchSnapshot()
  expect(map).not.toBeFalsy()
})

test('should not crash on an empty file', async () => {
  const bundle = await rollup({
    input: './test/gherkin.feature',
    plugins: [
      hypothetical({
        files: {
          './test/gherkin.feature': '',
        },
      }),
      rollupPluginGherkin(),
    ],
  })

  const { code, map } = await bundle.generate({ format: 'es', sourcemap: true })

  expect(code).toMatchSnapshot()
  expect(map).not.toBeFalsy()
})

test('should not output a sourcemap when disabled', async () => {
  const bundle = await rollup({
    input: './test/gherkin.feature',
    plugins: [
      hypothetical({
        files: {
          './test/gherkin.feature': `Feature: minimalism

  Scenario: minimalistic
    Given the minimalism
`,
        },
      }),
      rollupPluginGherkin({
        sourcemap: false,
      }),
    ],
  })

  const { map } = await bundle.generate({ format: 'es', sourcemap: false })

  expect(map).toBeFalsy()
})

test('should not touch a non-gherkin file', async () => {
  const bundle = await rollup({
    input: './test/gherkin.js',
    plugins: [
      hypothetical({
        files: {
          './test/gherkin.js': `console.log(42)
`,
        },
      }),
      rollupPluginGherkin(),
    ],
  })

  const { map } = await bundle.generate({ format: 'es', sourcemap: false })

  expect(map).toBeFalsy()
})
