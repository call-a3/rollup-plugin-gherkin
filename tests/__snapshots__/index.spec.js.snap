// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should compile Gherkin code correctly 1`] = `
"const pickles = [
  {
    tags: [],
    name: \\"minimalistic\\",
    language: \\"en\\",
    locations: [{\\"line\\":7,\\"column\\":3}],
    steps: [
      {\\"text\\":\\"the minimalism inside a background\\",\\"arguments\\":[],\\"locations\\":[{\\"line\\":4,\\"column\\":11}]},
      {\\"text\\":\\"the minimalism\\",\\"arguments\\":[],\\"locations\\":[{\\"line\\":8,\\"column\\":11}]},
      {\\"text\\":\\"a minimalist is minimalistic\\",\\"arguments\\":[],\\"locations\\":[{\\"line\\":9,\\"column\\":10}]},
      {\\"text\\":\\"minimalism is minimally achieved\\",\\"arguments\\":[],\\"locations\\":[{\\"line\\":10,\\"column\\":10}]}
    ]
  }
];

export default pickles;
"
`;

exports[`should not crash on an empty file 1`] = `
"const pickles = [
];

export default pickles;
"
`;
